### PR TITLE
docs: RFC for schema-registry-less CDL

### PR DESCRIPTION
**Pull request by frondeus**

### CODE REVIEW

File: docs/rfc/CDLF-00015-00-rfc-01.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+### Assumptions
+
+* CDL should allow to omit Schema Registry deployment and provide an alternative way of routing data.
+* Considering CIM format, any new fields or configuration must not be a breaking change.
+
+### Limitations
+* In this alternative, state routing would become a static table, which requires restarting the services after each change.
+* Data validation would be unavailable as we would not keep track of schema definitions.
+* Object versioning becomes abstract, almost meaningless without schema definitions.
```
Comment (Kononnable) [inline docs/rfc/CDLF-00015-00-rfc-01.md:92]: Not sure how object versions are related to schemas. Object can be changed even without changing its schema.

File: docs/rfc/CDLF-00015-00-rfc-01.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    object_id: UUID,
+    schema_id: UUID,
+    data: { any valid JSON }
+}
+```
+
+v1.0 input message format to accommodate schema-registry-less routing:
+```
+{
+    repository: String?, // Optional field
```
Comment (luke-biel) [inline docs/rfc/CDLF-00015-00-rfc-01.md:45]: I'm more inclined to put this in `options` object. Who knows what may be added here in the future.

File: docs/rfc/CDLF-00015-00-rfc-01.md (hunk 1 above)
Comment (frondeus) [reply docs/rfc/CDLF-00015-00-rfc-01.md:92]:
```suggestion
```

File: docs/rfc/CDLF-00015-00-rfc-01.md (hunk 2 above)
Comment (frondeus) [reply docs/rfc/CDLF-00015-00-rfc-01.md:45]: I'm not sure whether `repository` should live in `options`. It feels that it's as important as `schema_id`.

Comment (luke-biel) [reply docs/rfc/CDLF-00015-00-rfc-01.md:45]: That's a valid argument. Depends on clasiffication i guess. I'm worried that if it'll live on top of the object users may consider custom routing as something that should be happening on regular basis.

**Review by Kononnable — APPROVED** [review]

**Pull request merged**